lib/spack/spack/build_systems/intel.py

```diff
@@ -1241,6 +1247,33 @@ def configure_rpath(self):
             with open(compiler_cfg, 'w') as fh:
                 fh.write('-Xlinker -rpath={0}\n'.format(compilers_lib_dir))
 
+    # Need a copy of the auto_dispatch_options to be used in the subclasses
+    auto_dispatch_options = auto_dispatch_options
+
+    @run_after('install')
+    def configure_auto_dispatch(self):
+        if self._has_compilers:
+            if ('auto_dispatch=none' in self.spec):
+                return
+
+            compilers_bin_dir = self.component_bin_dir('compiler')
+
+            for compiler_name in 'icc icpc ifort'.split():
+                f = os.path.join(compilers_bin_dir, compiler_name)
+                if not os.path.isfile(f):
+                    raise InstallError(
+                        'Cannot find compiler command to configure '
+                        'auto_dispatch:\n\t' + f)
+
+                ad = []
+                for x in auto_dispatch_options:
+                    if 'auto_dispatch={0}'.format(x) in self.spec:
+                        ad.append(x)
+
+                compiler_cfg = os.path.abspath(f + '.cfg')
```

Comment:
This is generating a file in the original compilers directory. Would it be possible to instead do this in the staging directory?

Reply:
I may be misunderstanding but this is run after the install so the file should be going to the compiler directory.

Reply:
OK yeah I think *I* misunderstood: this is to customize the compiler components provided by this package? If so this is fine.